components/server/src/express-util.ts

```diff
@@ -171,11 +171,9 @@ export const takeFirst = (h: string | string[] | undefined): string | undefined
 };
 
 export function clientIp(req: express.Request): string | undefined {
-    const forwardedFor = takeFirst(req.headers["x-forwarded-for"]);
-    if (!forwardedFor) {
+    const clientIp = takeFirst(req.headers["x-client-ip-for-analytics"]);
```

Comment:
Is there a way to fix https://github.com/gitpod-io/gitpod/blob/dd0826cc0d3432a6b601c86760387ce0afc01038/components/proxy/conf/Caddyfile#L71

As it's seems to have the same value as `x-forwarded-for`.

Maybe we can use `x-real-ip` instead of `x-client-ip-for-analytics` in general?

Reply:
Yes, let's do this as we discussed.

https://github.com/gitpod-io/gitpod/pull/15752 will stop setting `X-Real-IP` in Caddy so that we can set it at the LB instead.

Once that is live we can deploy this change to start reading from the `X-Real-IP` header.